src/test/test_snap_mapper.cc

```diff
@@ -668,6 +678,82 @@ TEST_F(SnapMapperTest, MultiPG) {
   run();
 }
 
+// Check to_object_key against current format to detect accidental changes in encoding
+TEST_F(SnapMapperTest, CheckObjectKeyFormat) {
+  init(1);
+  // <object, test_raw_key>
+  std::vector<std::tuple<hobject_t, std::string>> object_to_object_key({
+      {hobject_t{"test_object", "", 20, 0x01234567, 20, ""},
+	  "OBJ_.1_0000000000000014.76543210.14.test%uobject.."},
+      {hobject_t{"test._ob.ject", "k.ey", 20, 0x01234567, 20, ""},
+	  "OBJ_.1_0000000000000014.76543210.14.test%e%uob%eject.k%eey."},
+      {hobject_t{"test_object", "", 20, 0x01234567, 20, "namespace"},
+	  "OBJ_.1_0000000000000014.76543210.14.test%uobject..namespace"},
+      {hobject_t{
+	  "test_object", "", std::numeric_limits<snapid_t>::max() - 20, 0x01234567,
+	    std::numeric_limits<int64_t>::max() - 20, "namespace"},
+	  "OBJ_.1_7FFFFFFFFFFFFFEB.76543210.ffffffffffffffec.test%uobject..namespace"}
+    });
+
+  for (auto &[object, test_object_key]: object_to_object_key) {
+    auto object_key = get_tester().to_object_key(object);
+    std::cout << object << " should be "
+	      << test_object_key << " is "
+	      << get_tester().to_object_key(object)
+	      << std::endl;
```

Comment:
IMHO, a successful run should be quiet.

Reply:
Agree. I will remove the log lines

Reply:
Well, it should probably print in the case of a failure.